File: Swift/프로그래머스/2/12909. 올바른 괄호/올바른 괄호.swift
import Foundation

func solution(_ s:String) -> Bool {    
    var arr:[Character] = []    
    for char in s {        
        if char == "(" {
            arr.append("(")
        } else if char == ")" {
            if arr.isEmpty {
                return false
            }
            arr.popLast()
        }    
    }            
    return arr.isEmpty
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    const arr = s.split("")    
    const stack = []
    
    if (arr[0] === ')') return false
    
    for(let char of arr) {        
        if(char === ')') {
            stack.pop() 
        }
        if(char === '(') {
            stack.push(char)            
        }        
    }    
    return !stack.length
}